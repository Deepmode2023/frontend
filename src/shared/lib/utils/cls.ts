const BREAK_TYPE = ["undefined", "false", "null"];

export const cls = (...classname: Array<any>): string => {
  return classname
    .reduce((acc, cls) => {
      if (BREAK_TYPE.includes("" + cls)) return acc;
      if (Array.isArray(cls)) return [...acc, ...cls];

      if (typeof cls === "object" && cls !== null && !Array.isArray(cls)) {
        return [
          ...acc,
          ...Object.keys(cls).filter((cls) => !cls.startsWith("__")),
        ];
      }

      return [...acc, cls];
    }, [])
    .join(" ");
};
