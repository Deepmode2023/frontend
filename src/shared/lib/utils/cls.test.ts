import { cls } from "./cls";
/* @ts-ignore */
import { expect, test, describe } from "bun:test";

describe("Describe cls function ", () => {
  const mergeList = ["cls1", "cls2", undefined, null];
  const anotherMergeList = ["cls3"];
  const styles = { container: "cls1", __chech: "hashed" };

  test("Checked returned class string with two array", () => {
    expect(cls(...mergeList, ...anotherMergeList)).toBe("cls1 cls2 cls3");
  });

  test("Checked when we pass just one class", () => {
    expect(cls("just-string-class")).toBe("just-string-class");
  });

  test("Checked when we pass undefined or null or false", () => {
    expect(cls(undefined)).toBe("");
    expect(cls(null)).toBe("");
    expect(cls(false)).toBe("");
  });

  test("Checked when we pass array without spread operators", () => {
    expect(cls(["cls1", "cls2"])).toBe("cls1 cls2");
  });

  test("Checked when we pass styles object", () => {
    expect(cls(styles)).toBe("container");
  });

  test("Snapshot cls", () => {
    expect(cls(styles)).toMatchSnapshot();
  });
});
