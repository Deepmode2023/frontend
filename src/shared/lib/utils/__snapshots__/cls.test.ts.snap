// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Describe cls function  Snapshot cls 1`] = `"container"`;
